[RFC] arch: arm: aarch32: make initial mpu configuration dynamic

kumarnac opened the pull request:
Remove const from mpu configuraton to make initial configuration
set during runtime.

Signed-off-by: Nachiketa Kumar <nachiketa.kumar@intel.com>

kumarnac commented:
Further to discussions  at https://github.com/zephyrproject-rtos/zephyr/discussions/35212  RFC.

najumon1980 commented:
hi All,
the change is for use case related with flexible code relocation feature of Zephyr on different memory location. The application code which need to relocate should be update with corresponding MPU attribute once relocated. So during pre-kernel initialization (soc_init), dynamically update arm_mpu_config based on relocation information

Following are few justification for this:
a) In our controller, there is no flash memory and both code/data are reside in SRAM
b) The Kernel startup code will update the SRAM region to read only, execute only, read-write etc. based on the arm_mpu_config by programming MPU
c) the arm_mpu_config is only can be access from Kernel/supervisory mode
d) Also any kernel mode program can update the MPU to change a read only region(eg: arm_mpu_config) to any other memory attribute. That mean if a kernel mode code compromised then there is no protection to system resources.

Suggestion:
use KConfig which will declared arm_mpu_config as const or not. Default it will be declared as const which serve MCU which use Flash memory. For MCU which use SRAM only can define KConfig accordingly if required to make the arm_mpu_config modifiable from kernel mode code such soc_init.

kumarnac committed:
arch: arm: aarch32: make initial mpu configuration dynamic

Remove const from mpu configuraton to make initial configuration
set during runtime.

Signed-off-by: Nachiketa Kumar<nachiketa.kumar@intel.com>
